assets/large_language_models/rag/components/src/flow_creation.py

```diff
@@ -257,14 +330,67 @@ def main(args, ws, current_run, activity_logger: Logger):
         "@@Completion_Deployment_Name", completion_deployment_name)
     flow_with_variants = flow_with_variants.replace(
         "@@Completion_Connection", completion_connection_name)
+    flow_with_variants = flow_with_variants.replace(
+        "@@MLIndex_Asset_Id", mlindex_asset_id)
+
     api_name = "chat" if completion_model_name == "gpt-35-turbo" else "completion"
     flow_with_variants = flow_with_variants.replace("@@API", api_name)
 
-    activity_logger.info(
-        "[Promptflow Creation]: Json payload successfully generated, trying to parse into json dict...")
-    json_payload = json.loads(flow_with_variants)
-    activity_logger.info(
-        "[Promptflow Creation]: Json payload successfully parsed, submit to promptflow service now...")
+    if code_first:
+
+        # write flow dag yaml back to file
+        with open(os.path.join(Path(__file__).parent.absolute(), CODE_DIR, "flow.dag.yaml"), "w") as file:
+            file.write(flow_with_variants)
+        import codecs
+        # Write prompt file content for Variants
+        for idx in range(0, len(top_prompts)):
+            prompt_str = post_processing_prompts(
+                json_stringify(top_prompts[idx]),
+                _CITATION_TEMPLATE, _USER_INPUT, is_chat)
+            with open(os.path.join(
+                    Path(__file__).parent.absolute(),
+                    CODE_DIR,
+                    f"Prompt_variants__Variant_{idx}.jinja2"), "w") as file:
+                file.write(codecs.decode(prompt_str, 'unicode_escape'))
+
+        # upload code
+        yaml_path = upload_code_files(ws, flow_name) + "/flow.dag.yaml"
+
+        # Load in Json
+        json_name = os.path.join("flow_jsons", prefix + "flow_with_variants_mlindex_code_first.json")
+        with open(os.path.join(Path(__file__).parent.absolute(), json_name), "r") as file:
+            flow_submit_data = file.read()
+
+        flow_submit_data = flow_submit_data.replace(
+            "@@Flow_Name", flow_name)
+        flow_submit_data = flow_submit_data.replace(
+            "@@Flow_Definition_Path", yaml_path)
+        # replace values as it should
+        activity_logger.info(
```

Comment:
Add logs to denote that this is code first flow. Otherwise the messages are same as the ones on line 389-393

Reply:
added